src/System.Runtime.Intrinsics.X86/src/System/Runtime/Intrinsics/X86/Avx.cs

```diff
@@ -0,0 +1,577 @@
+        // __m256i _mm256_cvttps_epi32 (__m256 a)
+        public static Vector256<int> ConvertToVector256IntWithTruncation(Vector256<float> value) { throw new NotImplementedException(); }
+
+        // __m256 _mm256_div_ps (__m256 a, __m256 b)
+        public static Vector256<float> Divide(Vector256<float> left, Vector256<float> right) { throw new NotImplementedException(); }
+        // __m256d _mm256_div_pd (__m256d a, __m256d b)
+        public static Vector256<double> Divide(Vector256<double> left, Vector256<double> right) { throw new NotImplementedException(); }
+
+        // __m256 _mm256_dp_ps (__m256 a, __m256 b, const int imm8)
+        public static Vector256<float> DotProduct(Vector256<float> left, Vector256<float> right, byte control) { throw new NotImplementedException(); }
+        
+        // __m256 _mm256_moveldup_ps (__m256 a)
+        public static Vector256<float> DuplicateEvenIndexed(Vector256<float> value) { throw new NotImplementedException(); }
+        // __m256d _mm256_movedup_pd (__m256d a)
+        public static Vector256<double> DuplicateEvenIndexed(Vector256<double> value) { throw new NotImplementedException(); }
+
+        // __m256 _mm256_movehdup_ps (__m256 a)
+        public static Vector256<float> DuplicateOddIndexed(Vector256<float> value) { throw new NotImplementedException(); }
+
+
+        // __int8 _mm256_extract_epi8 (__m256i a, const int index)
+        public static sbyte ExtractSbyte<T>(Vector256<T> value, byte index) where T : struct { throw new NotImplementedException(); }
+        // __int8 _mm256_extract_epi8 (__m256i a, const int index)
+        public static byte ExtractByte<T>(Vector256<T> value, byte index) where T : struct { throw new NotImplementedException(); }
+        // __int16 _mm256_extract_epi16 (__m256i a, const int index)
+        public static short ExtractShort<T>(Vector256<T> value, byte index) where T : struct { throw new NotImplementedException(); }
+        // __int16 _mm256_extract_epi16 (__m256i a, const int index)
+        public static ushort ExtractUshort<T>(Vector256<T> value, byte index) where T : struct { throw new NotImplementedException(); }
+        // __int32 _mm256_extract_epi32 (__m256i a, const int index)
+        public static int ExtractInt<T>(Vector256<T> value, byte index) where T : struct { throw new NotImplementedException(); }
+        // __int32 _mm256_extract_epi32 (__m256i a, const int index)
+        public static uint ExtractUint<T>(Vector256<T> value, byte index) where T : struct { throw new NotImplementedException(); }
+        // __int64 _mm256_extract_epi64 (__m256i a, const int index)
+        public static long ExtractLong<T>(Vector256<T> value, byte index) where T : struct { throw new NotImplementedException(); }
+        // __int64 _mm256_extract_epi64 (__m256i a, const int index)
+        public static ulong ExtractUlong<T>(Vector256<T> value, byte index) where T : struct { throw new NotImplementedException(); }
+        
```

Comment:
Pattern ExtractXxx&lt;T&gt; could be problematic as generic parameter restriction to struct does not eliminate invalid method uses i.e. ExtractInt&lt;RSAParameters&gt; - this syntax is valid with  current definition but it is invalid at runtime. This is a general problem which pops up in all cases where generic parameter in method definition is restricted to struct only.